### PR TITLE
watch default secrets

### PR DESCRIPTION
**Pull request by maksymvavilov**
Improve watch to queue records with no provider ref on update event to the default provider secret.
Also, updating a test to set provider ref to `nil` instead of an empty string to test for #500 bug

### CODE REVIEW

**Review by mikenairn — COMMENTED** [review]
I think the changes look fine, just a question around the check, but either way it will work as required.

File: internal/controller/dnsrecord_controller.go [hunk 1]
```diff
@@ -449,11 +449,19 @@ func (r *DNSRecordReconciler) SetupWithManager(mgr ctrl.Manager, maxRequeue, val
...
+			isDefaultSecret := s.Labels[v1alpha1.DefaultProviderSecretLabel] == "true"
+
 			for _, record := range records.Items {
 				if record.Status.ProviderRef.Name == o.GetName() {
 					logger.Info("secret updated", "secret", o.GetNamespace()+"/"+o.GetName(), "enqueuing dnsrecord ", record.GetName())
 					toReconcile = append(toReconcile, reconcile.Request{NamespacedName: client.ObjectKeyFromObject(&record)})
 				}
+
+				// if this is a default secret also reconcile records without secret ref
+				if isDefaultSecret && record.Spec.ProviderRef == nil {
```
[the remark this answers is not in the document]
Comment (philbrookes) [reply internal/controller/dnsrecord_controller.go:462]: this makes sense to me if it's an or rather than an and as suggested. As an and it can't be true that `providerRef` is nil and `providerRef.Name` is `""`.

I'd think:
```
if isDefaultSecret && (record.Spec.ProviderRef == nil || record.Spec.ProviderRef != nil && record.Status.ProviderRef.Name == "")
```

[the remark this answers is not in the document]
Comment (maksymvavilov) [reply internal/controller/dnsrecord_controller.go]: A good point - we also need to check for a secret not being assigned. 
I went with the 
```
defaultSecret && ( Spec.ProviderRef == nil || Spec.ProfiderRef.Name == nil) && Status.ProviderRef == ""
```
to further resrtict as Mike suggested and included the `||` to not only ceck for nil but also for an empty ref (mirror the [code](https://github.com/Kuadrant/dns-operator/blob/gh-499/internal/controller/dnsrecord_controller.go#L223) in the controller)

**Review by philbrookes — APPROVED** [review]
